packages/headless/src/api/platform-client.ts

```diff
@@ -124,76 +124,55 @@ export class PlatformClient {
   }
 }
 
-interface URLOptions<E extends PlatformEnvironment> {
-  environment?: E;
-  region?: Extract<PlatformCombination, {env: E}>['region'];
-  multiRegionSubDomain?: string;
-}
-
-function coveoCloudURL<E extends PlatformEnvironment>(
-  subdomain: string,
-  options?: URLOptions<E>
+export function getDefaultOrganizationEndpointBaseUrl(
```

Comment:
The reason why I did it this way is we typically only need one of the values in any given place in the code, but I don't really mind returning them all every time TBH.

As for adding `search` there, I'm not a fan. This endpoint will never be used in the commerce engine, so returning it will always be pointless in that context.

Reply:
> As for adding search there, I'm not a fan. This endpoint will never be used in the commerce engine, so returning it will always be pointless in that context.

IMO the platform-client, shouldn't have to worry about the fact that the commerce engine doesn't use the search endpoint. There should be an added layer on top of platform-client that is `commerce-client` and that one doesn't use the `search` baseUrl but instead will define it's own `commerce` baseUrl and simply not use the `search` one.

I feel like search is pretty core to what we do that it's okay for it to be part of the platform-client. Even if layers on top : `commerce`, `service`, don't always need it.

Reply:
I tried reverting to a function that returns all org endpoints... and honestly I don't like it.

As I mentioned earlier, in almost every place in the code (except maybe in some tests), we only need to get one of the three org endpoints at a time. Getting them all every time feels somewhat clumsy for that reason.

Instead of passing a parameter to get just the endpoint you need, you end up getting three endpoints, two of which will be useless, and selecting the one you need from there.

E.g., if I need the analytics endpoint, it's not any harder or more error-prone to do:

```typescript
const analyticsOrgEndpoint = getOrganizationEndpoint(organizationId, environment, 'analytics');
```

than it is to do:

```typescript
const analyticsOrgEndpoint = getOrganizationEndpoints(organizationId, environment).analytics
```

but IMO the former feels more "right".